src/components/Modal/Modal.tsx

```diff
@@ -256,6 +271,7 @@ export const Modal = forwardRef<HTMLDivElement, ModalProps>(function Modal(
                       aria-describedby={subtitleId}
```

Comment:
we are loosing the aria-describedby and aria-labelledby functionality with the refactor, do you think it would be worth adding that it for this pr?

Reply:
ohh good catch. Do you think using an H2 and H3 in modal title would suffice?

Reply:
Nvm, gonna use context to pass the aria description and label ids!

Reply:
yeah context is perf :)